server/src/watchdog.js

```diff
@@ -103,6 +103,12 @@ exports.submit = function(shot) {
     return;
   }
 
+  if (config.watchdog.devOnlyMatchHostname
+      && ((new URL(shot.origin)).hostname !== config.watchdog.devOnlyMatchHostname)) {
```

Comment:
Wrap the `new URL()` call in a try/catch to prevent errors due to shots from weird URLs like `file:///`, where `siteName` is set to `null` when uploading ([here's](https://screenshots.firefox.com/WLyju7NyRbn6jwGs/null) an example).